### DISCUSSION TITLE
@next/bundle-analyzer does not automatically handle config function

### DISCUSSION THREAD

**Discussion by vinverdy** [vercel/next.js#44637] [Help]
### Describe the feature you'd like to request

we usually handle next.config.js like this

`module.exports = withBundleAnalyzer(withX(withY(configObject)))`

if at some point withSomething return config function instead of config object, our config will be lost. i realize on the [docs](https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer) mentioned to use specific way to handle config function, but i expect handle it automatically

### Describe the solution you'd like

withBundleAnalyzer should handle whether the config is object or function, and handle internally

### Describe alternatives you've considered

cant think other option

**Comment by SukkaW** [comment]
IMHO it is because you are using it in a wrong way.

The correct way of using `@next/bundle-analyzer` is:

```js
module.exports = withBundleAnalyzer()(withX(withY(configObject)))
```
